modules/face/samples/sampleKazemi.cpp

```diff
@@ -0,0 +1,106 @@
+#include<iostream>
+#include<vector>
+#include<fstream>
+#include "opencv2/face/face_alignment.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+using namespace std;
+using namespace cv;
+int main(int argc,char** argv){
+    //Give the file containing the filenames of filenames as argument
+    if(argc!=2){
+        cout<<"No file found";
+        return 0;
+    }
+    string file = argv[1];
+    //cout<<file<<endl;
+    ifstream f(file.c_str());
+    if(!f.is_open()){
+        cout<<"Error while loading file";
+        return 0;
+    }
+    //create a vector to store names of files in which annotations
+    // and image names are found
+    /*The format of the file containing annotations should be of following format
+        /data/abc.jpg
+        123.45,345.65
+        321.67,543.89
+     */
+    vector<string> filenames;
+    string s;
+    while(getline(f,s)){
+        //cout<<s<<endl;
+        filenames.push_back(s);
+    }
+    f.close();
+    //create a vector to store image names
+    vector<string> imagenames;
+    //create object to get landmarks
+    vector<vector<Point2f> > trainlandmarks;
+    Ptr<cv::face::FacemarkKazemi> facemark=cv::face::create();
+    //gets landmarks and corresponding image names in both the vectors
+    facemark->getData(filenames,trainlandmarks,imagenames);
+    //vector to store images
+    vector<Mat> trainimages;
+    for(unsigned long i=0;i<imagenames.size();i++){
+        //cout<<imagenames[i]<<endl;
+        char b[100];
+        unsigned long z=0;
+        while(imagenames[i][z]!='\n'&&imagenames[i][z]!='\r'&&z<imagenames[i].length()){
+            b[z]=imagenames[i][z];
+            z++;
+        }
+        string img =string(b);
+        Mat src = imread(b);
+        if(!src.data){
+            continue;
+        }
+        trainimages.push_back(src);
+    }
+    //scale data according to a particular size
+    facemark->scaleData(trainlandmarks,trainimages,Size(200,200));
```

Comment:
This can be any size. It is just the reference at which all landmarks can be scaled to calculate the mean. I have created a seperate function getMeanShaperelative which calculates the mean shape relative to the detected bounding rectangle by warping these points. I have just tried to remove the overhead of calculating mean again and again. We can initially calculate mean to any reference and then warp that according to different images according to faces detected.